src/auth/auth.service.ts

```diff
@@ -28,16 +28,19 @@ export class AuthService {
         return this.sign(user);
     }
 
-    async validateUser({ email, password }: AuthCredentialsDTO): Promise<User> {
-        const user: User = await this.userService.findOne({ email }, true);
+    async validateUser({
+        username,
+        password,
+    }: AuthCredentialsDTO): Promise<User> {
+        const user: User = await this.userService.findOne({ username }, true);
```

Comment:
I see you use only `username` to retrieve user here? What's the role of a password here?

Reply:
To compare whether input password is correct or not. See the code just below this one.